File: Demo/OC/AppFrameworks/UIKit/YJUIPageViewManager/YJUIPageViewManager/ViewController.m
//
//  ViewController.m
//  YJUIPageViewManager
//
//  Created by admin on 2017/5/3.
//  Copyright © 2017年 YJCocoa. All rights reserved.
//

#import "ViewController.h"
#import "YJUIPageViewManager-Swift.h"
#import <YJCocoa/YJCocoa-Swift.h>

@interface ViewController ()

@property (nonatomic, strong) YJUIPageViewController *pageView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self initUI];
    [self testClass];
}

- (void)initUI {
    self.pageView = [[YJUIPageViewController alloc] initWithTransitionStyle:UIPageViewControllerTransitionStyleScroll navigationOrientation:UIPageViewControllerNavigationOrientationHorizontal options:nil];
    self.pageView.view.frame = self.view.bounds;
    [self.view addSubview:self.pageView.view];
    [self addChildViewController:self.pageView];
}

- (void)testClass {
    self.pageView.manager.isDisableBounces = YES; // 关闭阻力效果
    // 填充数据源
    NSMutableArray *array = NSMutableArray.array;
    for (int i=0; i<5; i++) {
        [array addObject:[YJTestPageViewCell cellObject]];
    }
    self.pageView.dataSourceCell = array;
    [self.pageView reloadData];
}

@end

File: Demo/Swift/AppFrameworks/UIKit/YJPageView/YJPageView/ViewController.swift
//
//  ViewController.swift
//  YJPageView
//
//  Created by 阳君 on 2019/5/24.
//  Copyright © 2019 YJCocoa. All rights reserved.
//

import UIKit
import YJCocoa

class ViewController: UIViewController {
    
    lazy var pageVC: YJUIPageViewController! = {
        let pageVC = YJUIPageViewController(transitionStyle: .scroll, navigationOrientation: .horizontal, options: nil)
        pageVC.view.frame = self.view.bounds
        pageVC.manager.isLoop = false
        pageVC.manager.isDisableBounces = true
        return pageVC
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.addChild(self.pageVC)
        self.view.addSubview(self.pageVC.view)
        self.testData()
    }
    
    func testData() {
        for _ in 0..<10 {
            self.pageVC.dataSourceCell.append(YJTestPVCell.cellObject())
        }
        self.pageVC.reloadData()
    }
    
}
